### DISCUSSION TITLE
Coding Rate for the CDP

### DISCUSSION THREAD

**Discussion by tqrahman** [Code-and-Response/ClusterDuck-Protocol#329] [QuackOverflow] [answered]
What is the coding rate for CDP and where Is it defined? I looked at the cdpcfg.h file and seems not to be defined there.

**Comment by amirna2** [comment, marked as answer]
The coding rate isn't explicitly defined like the other Radio parameters in DuckRadio.cpp.
As a result it is set to a default value in the RadioLib `begin()` method.

If you want to define the parameter, you can either choose a CDP specific value and explicitly call the API, like below or
you can use the `begin(....)` method and pass it radio parameters.
I prefer the below mechanism so each parameters is set individually as I have seen some timing issues in RadioLib when all parameters are set from the `begin` method.

```
  rc = lora.setCodingRate(7);
  if (rc == RADIOLIB_ERR_INVALID_CODING_RATE) {
    logerr("ERROR  coding rate is invalid");
    return DUCKLORA_ERR_SETUP;
  }
  ```
  
  Possible values from RadioLib are
  ```
#define RADIOLIB_SX126X_LORA_CR_4_5                            0x01        //  7     0     LoRa coding rate: 4/5
#define RADIOLIB_SX126X_LORA_CR_4_6                            0x02        //  7     0                       4/6
#define RADIOLIB_SX126X_LORA_CR_4_7                            0x03        //  7     0                       4/7
#define RADIOLIB_SX126X_LORA_CR_4_8                            0x04        //  7     0                       4/8
```

**Reply by tqrahman** [reply]
Thank you @amirna2 for the explanation and a detailed code on how to set the coding rate!
